src/sourmash/tax/__main__.py

```diff
@@ -0,0 +1,300 @@
+
+
+def classify(args):
+    """
+    taxonomic classification of genomes from gather results
+    """
+    set_quiet(args.quiet)
+
+    # first, load taxonomic_assignments
+    tax_assign = {}
+    available_ranks = set()
+    for tax_csv in args.taxonomy_csv:
+
+        try:
+            this_tax_assign, _, avail_ranks = tax_utils.load_taxonomy_csv(tax_csv, split_identifiers=not args.keep_full_identifiers,
+                                              keep_identifier_versions = args.keep_identifier_versions,
+                                              force=args.force)
+            # to do -- maybe check for overlapping tax assignments? rn later ones will override earlier ones
+            tax_assign.update(this_tax_assign)
+            available_ranks.update(set(avail_ranks))
+        except ValueError as exc:
+            error(str(exc))
+
+    if not tax_assign:
+        error(f'No taxonomic assignments loaded from {args.taxonomy_csv}. Exiting.')
+        sys.exit(-1)
+
+    if args.rank and args.rank not in available_ranks:
+        error(f"No taxonomic information provided for rank {args.rank}: cannot classify at this rank")
+        sys.exit(-1)
+
+    # get gather_csvs from args
+    gather_csvs = tax_utils.collect_gather_csvs(args.gather_csv, from_file=args.from_file)
+
+    classifications = defaultdict(list)
+    matched_queries=set()
+    krona_results = []
+    num_empty=0
+    status = "nomatch"
+    seen_perfect = set()
+
+    # handle each gather result separately
+    for n, g_csv in enumerate(gather_csvs):
+        gather_results, idents_missed, total_missed, _ = tax_utils.check_and_load_gather_csvs(g_csv, tax_assign, force=args.force,
+                                                                                 fail_on_missing_taxonomy=args.fail_on_missing_taxonomy)
+
+        if not gather_results:
+            num_empty += 1
+            continue
+
+        # if --rank is specified, classify to that rank
+        if args.rank:
+            best_at_rank, seen_perfect = tax_utils.summarize_gather_at(args.rank, tax_assign, gather_results, skip_idents=idents_missed,
+                                                         split_identifiers=not args.keep_full_identifiers,
+                                                         keep_identifier_versions = args.keep_identifier_versions,
+                                                         best_only=True, seen_perfect=seen_perfect)
+
+           # this now returns list of SummarizedGather tuples
+            for (query_name, rank, fraction, lineage) in best_at_rank:
+                status = 'nomatch'
+                if query_name in matched_queries:
+                    continue
+                if fraction <= args.containment_threshold:
+                    status="below_threshold"
+                    notify(f"WARNING: classifying at desired rank {args.rank} does not meet containment threshold {args.containment_threshold}")
```

Comment:
does this give enough information to figure out what entry is problematic? maybe include `query_name` in the error output?